Template for decommissioning a linux server (#6)
idsys-unibe-ch/.github: .github/ISSUE_TEMPLATE/decommissioning_v_1_0.md

# Decommissioning a Linux Server  

> **Describe wich and why will be a server decommissioning**

Reviewer: ```suggestion
**Describe which server(s) is/are to be decommissioned and why**
```

> <p>ID-Service Now: <input type="text" size="20" maxlength="40" name="ID-Number" value="ID-Number" /></p>  <!-- put the ID Number here -->

Reviewer: Do these html input tags actually work? Why use them at all? As all repos "should" have autoreferencing Service now tickets, I'd suggest to simply provide a hint regarding the syntax wherever a possible Service now ticket ID is helpful:

```suggestion

ID-Service Now: SNOW-IDxxx

```

Author: I tested this code with the markdown preview and it looked ok (see image). 
Unfortunately, how the template ultimately looks like (using this codes) can only be seen by applying the template.

If you agree, I would like to leave this code and test it. If it doesn't work then I'll apply your proposal.

![Bildschirmfoto 2024-09-13 um 12 09 55](https://github.com/user-attachments/assets/51398144-148d-474d-aba8-0f0ee2893357)

Reviewer: ok, let‘s see what happens.

<!-- A clear and concise description of what the problem is. -->  

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Additional context**
<!-- Add any other context about the problem here. -->

> # Workflow

Reviewer: In the issue type *task* we call this section action items.

```suggestion
# Action Items
```

<!--The steps from confluence https://unibe-ch.atlassian.net/wiki/x/moU8Aw  are followed.  -->

>**Servername :** <!-- Servername to decommission -->  
>**FQDN:** <!-- FQDN -->  
>**IP:** <!-- IP -->  
>**Contact person:** <!-- Contact person -->  

> >**Projectname:** <!-- Name of the project -->  

Reviewer: I'd put these system facts in the first general information part above (directly after the first heading?) and only leave action items in this section.

> - [ ] System in PRTG stop with a comment (ID-Service Now)

Reviewer: ```suggestion
- [ ] Pause system in [PRTG](https://idprtg.unibe.ch) indefinitely with a comment "marked for decommissioning"
```

- [ ] Firewall-Rules: create a Ticket in Service Now to assignment group "ID - Security" (Technical Service: Firewalldienste) with Hostname/IP with the request to delete all firewall rules 
       <p>ID-Service Now: <input type="text" size="20" maxlength="40" name="ID-Number" value="ID-Number" /></p>   <!-- put the ID Number here -->
- [ ] Elasticsearch: create a Ticket in Service Now to assignment "ID - Security" (Technical Service: Elasticsearch) with the request to delete the host from elasticsearch. 
        <p>ID-Service Now: <input type="text" size="20" maxlength="40" name="ID-Number" value="ID-Number" /></p> <!-- put the ID Number here -->
- [ ] Check documentation to see if there are any references to other dependencies that need to be deleted
- [ ] Delete any documentation for the System
- [ ] Inform those responsible for the system and ask them to delete relevant documentation in their space in confluence.
- [ ] Database in PostgreSQL will also be deleted if it exists. 
- [ ] NAS configs will be deleted. Send a Email with the Servername and IP to Roland, Trummer <roland.trummer@unibe.ch> and wait the confirmation

> - [ ]  Shut des Systems:

Reviewer: ```suggestion
- [ ] Shut down the system
```

> > - [ ] - a.) If ESXI VM cannot be deleted → create a Ticket to ESXI Team with the request to delete the server
> > - [ ] - b.) If hardware system → remove the disks, wipe them and pass them on correctly place (ESXI Büro in die Blaue Box), Check the inventory number and check the next steps with Lead SYS (disposal / Storage)
> - [ ] Delete Server

Reviewer: ```suggestion
- [ ] Delete the server:
  - [ ] a.) delete VM in vCenter, if VM cannot be deleted → create a Ticket to ESXI Team with the request to delete the server
  - [ ] b.) If hardware system → remove the disks, wipe them and pass them on correctly place (ESXI office into the blue box), Check the inventory number and check the next steps with Lead SYS (disposal / Storage)

```

Not sure if the blue box is still correct. Could also be elsewhere nowadays (office WPS, cellar, ..?)

Author: I asked Roland Trummer. There is no longer a blue box. If the disk has been wiped, dispose of it in the container in the basement, where the electronic waste is.
